internal/ratelimit/handler.go

```diff
@@ -114,10 +116,29 @@ func extractResourceAction(path, method string) (res, act string, err error) {
 // set to the number of seconds the client should wait to make it's next request.
 func Handler(l *rate.Limiter, next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		path := req.URL.Path
-		method := req.Method
+		reqInfo, ok := event.RequestInfoFromContext(req.Context())
+		if !ok || reqInfo == nil || reqInfo.ClientIp == "" {
+			// TODO: what to do here?
+			// We could:
+			// 1. Ignore and continue ahead, with the ip being empty...
+			// 2. Have fallback logic to extract the Ip
+			// 3. Return an error since it would mean the middlewares are not setup correctly.
+			rw.WriteHeader(http.StatusInternalServerError)
+			return
+		}
 
-		res, a, err := extractResourceAction(path, method)
+		authtoken, ok := req.Context().Value(globals.ContextAuthTokenPublicIdKey).(string)
+		if !ok {
```

Comment:
I removed the `authtoken == ""` check, since this gets set to an empty string for unauthed requests like for the request to `/v1/auth-methods/{id}:authenticate`.